Common/Data/Auxiliary/FactorFileZipHelper.cs

```diff
@@ -0,0 +1,76 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Linq;
+using QuantConnect.Util;
+using System.Collections.Generic;
+
+namespace QuantConnect.Data.Auxiliary
+{
+    /// <summary>
+    /// Provides methods for reading factor file zips
+    /// </summary>
+    public static class FactorFileZipHelper
+    {
+        /// <summary>
+        /// Reads the zip bytes as text and parses as FactorFileRows to create FactorFiles
+        /// </summary>
+        public static Dictionary<Symbol, FactorFile> ReadFactorFileZip(Stream file, MapFileResolver mapFileResolver, string market)
+        {
+            if (file == null || file.Length == 0)
+            {
+                return new Dictionary<Symbol, FactorFile>();
+            }
+
+            var dict = (
+                    from kvp in Compression.Unzip(file)
+                    let filename = kvp.Key
+                    let lines = kvp.Value
+                    let factorFile = SafeRead(filename, lines)
+                    let mapFile = mapFileResolver.GetByPermtick(factorFile.Permtick)
+                    where mapFile != null
+                    let sid = SecurityIdentifier.GenerateEquity(mapFile.FirstDate, mapFile.FirstTicker, market)
+                    let symbol = new Symbol(sid, mapFile.Permtick)
+                    select new { Symbol = symbol, FactorFile = factorFile }
+                )
+                .DistinctBy(x => x.Symbol)
+                .ToDictionary(x => x.Symbol, x => x.FactorFile);
+            file.DisposeSafely();
```

Comment:
`file.DisposeSafely()` think it's better that this is handled by the caller which is requesting the stream, same for `MapFileZipHelper`